src/inmanta/server/server.py

```diff
@@ -188,116 +72,8 @@ def get_depended_by(self) -> List[str]:
-        :return: The logger for the given environment.
-        """
-        if environment in self._resource_action_loggers:
-            return self._resource_action_loggers[environment]
-
-        resource_action_log = self.get_resource_action_log_file(environment)
-
-        file_handler = logging.handlers.WatchedFileHandler(filename=resource_action_log, mode="a+")
-        # Most logs will come from agents. We need to use their level and timestamp and their formatted message
-        file_handler.setFormatter(logging.Formatter(fmt="%(message)s"))
-        file_handler.setLevel(logging.DEBUG)
-
-        resource_action_logger = logging.getLogger(const.NAME_RESOURCE_ACTION_LOGGER).getChild(str(environment))
-        resource_action_logger.setLevel(logging.DEBUG)
-        resource_action_logger.addHandler(file_handler)
-
-        self._resource_action_loggers[environment] = resource_action_logger
-        self._resource_action_handlers[environment] = file_handler
-
-        return resource_action_logger
-
-    def _close_resource_action_loggers(self) -> None:
-        """Close all resource action loggers and their associated handlers"""
-        try:
-            while True:
-                env, logger = self._resource_action_loggers.popitem()
-                self._close_resource_action_logger(env, logger)
-        except KeyError:
-            pass
-
-    def _close_resource_action_logger(self, env: uuid.UUID, logger: logging.Logger = None) -> None:
-        """Close the given logger for the given env.
-        :param env: The environment to close the logger for
-        :param logger: The logger to close, if the logger is none it is retrieved
-        """
-        if logger is None:
-            if env in self._resource_action_loggers:
-                logger = self._resource_action_loggers.pop(env)
-            else:
-                return
-
-        handler = self._resource_action_handlers.pop(env)
-        logger.removeHandler(handler)
-        handler.flush()
-        handler.close()
-
-    def log_resource_action(
-        self, env: uuid.UUID, resource_ids: List[str], log_level: int, ts: datetime.datetime, message: str
-    ) -> None:
-        """Write the given log to the correct resource action logger"""
-        logger = self.get_resource_action_logger(env)
-        if len(resource_ids) == 0:
-            message = "no resources: " + message
-        elif len(resource_ids) > 1:
-            message = "multiple resources: " + message
-        else:
-            message = resource_ids[0] + ": " + message
-        log_record = ResourceActionLogLine(logger.name, log_level, message, ts)
-        logger.handle(log_record)
-
-    def get_agent_client(self, tid: UUID, endpoint: str) -> Optional[protocol.ReturnClient]:
-        return self.agentmanager.get_agent_client(tid, endpoint)
+        self.agentmanager: "AgentManager" = cast("AgentManager", server.get_slice(SLICE_AGENT_MANAGER))
```

Comment:
The SLICE_AGENT_MANAGER and the SLICE_COMPILER service are used here, but they are not part of the dependency list.

Reply:
done